testrunner-running/src/main/scala/edu/illinois/cs/testrunner/execution/Utils.java

```diff
@@ -4,27 +4,46 @@
 import java.util.regex.Pattern;
 
 public class Utils {
-    private static final String junit5Regex = "\\[class:([\\w.]+).*\\[method:(\\w+)\\(";
-    private static final Pattern junit5Pattern = Pattern.compile(junit5Regex);
+    private static final String junit5NestedRegex = "\\[nested-class:([\\w.]+)\\]";
+    private static final Pattern junit5NestedPattern = Pattern.compile(junit5NestedRegex);
+    private static final String junit5BasicRegex = "\\[class:([\\w.]+).*\\[method:([\\w().]+)";
+    private static final Pattern junit5BasicPattern = Pattern.compile(junit5BasicRegex);
     private static final String junit4Regex = "\\[test:(\\w+)\\(([\\w.]+)\\)";
     private static final Pattern junit4Pattern = Pattern.compile(junit4Regex);
 
     /**
      * Turn the uniqueId from identifier into fully qualified method name.
      *
+     * For JUnit 5 nested test:
+     * uniqueId: [engine:junit-jupiter]/[class:com.luojl.demo.InheritedTest]/[nested-class:NestedTest]/[method:NestedTestB()]
+     * fully qualified name: com.luojl.demo.InheritedTest$NestedTest#NestedTestB()
+     *
      * For JUnit 5:
      * uniqueId: [engine:junit-jupiter]/[class:com.luojl.demo.JUnit5DemoTest]/[method:TestC()]
-     * full qualified name: com.luojl.demo.JUnit5DemoTest#TestC
+     * full qualified name: com.luojl.demo.JUnit5DemoTest#TestC()
```

Comment:
To double-check, do we want the () at the end? Is this what JUnit 5's runner expects?

Reply:
Yes, we want the (). And this is also for the parameterized tests, which will be the next PR.

There are many different kinds of fully qualified name, refer to the example in this [page](https://junit.org/junit5/docs/5.0.0/api/org/junit/platform/launcher/core/LauncherDiscoveryRequestBuilder.html) (Check the content in the `selectors`)